### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by JavaEria**
Currently working on filter function and also improving naming conventions.

### CODE REVIEW

File: src/EditTodoItem.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import React from "react";
+import './EditTodoItem.css'
+
+class EditTodoItem extends React.Component {
+  
+  constructor(props) {
+    super(props)
```
Comment (mashhoodr) [inline src/EditTodoItem.js:6]: Please configure linting in this repo

File: src/App.js [hunk 2]
```diff
@@ -3,12 +3,13 @@ import logo from './logo.svg';
...
 import InputTodo from './InputTodo';
 import TodoList from './TodoList';
+import FilterTodo from './FilterTodo'
 
 class App extends Component {
 
   constructor(props) {
     super(props)
-    this.state = {data: []}
+    this.state = {data: [], filteredData: [], isFiltered: false}
```
Comment (mashhoodr) [inline src/App.js]: do we need to keep both data and filtered data in state? The other one can be just stored in class.

Comment (JavaEria) [reply src/App.js]: We were using filteredData for the TodoList component when the filtering is required, as we could not manipulate the data array on every filter. Should we look for an alternate way for this?

Comment (JavaEria) [reply src/App.js]: We have changed our approach, we have removed filteredData from the state and are using it as a class object in the TodoList component

File: src/InputTodo.js [hunk 3]
```diff
@@ -4,20 +4,19 @@ import uuid from'react-native-uuid';
 class InputTodo extends Component {
```
Comment (QaseemLodhi) [inline src/InputTodo.js:4]: why are we importing an entire separate lib for generating simple random id? -sa

File: src/InputTodo.js [hunk 4]
```diff
@@ -4,20 +4,19 @@ import uuid from'react-native-uuid';
...
-       return {id: uuid.v1(), name: todoDescription, isChecked: false};
+       return {id: uuid.v1(), name: todoDescription, isChecked: false, isEditable: false};
     }
     
 
     sendTodo = (event) => {
         event.preventDefault();
-        this.props.recieveTodo(this.createTodo(event.target.elements.inputItem.value));
-
+        this.props.recieveTodo(this.createTodo(event.target.elements.inputTodo.value));
```
Comment (QaseemLodhi) [inline src/InputTodo.js:13]: we can store "event.target.elements.inputTodo.value" into a seperate local variable and then pass to function and so on, -qw
let description = event.target.elements.inputTodo.value;
this.props.recieveTodo(this.createTodo(description));

File: src/App.js [hunk 5]
```diff
@@ -1,44 +1,62 @@
...
+    this.state = {data: [], filterName: ''}
   }
-  
-  recieveTodo = (item) => {
+
+  recieveTodo = item => {
     this.state.data.push(item);
-    this.setState({data: this.state.data});
-  }
+    this.setState({ data: this.state.data });
```
Comment (QaseemLodhi) [inline src/App.js]: whats the point of directly manipulating component state and then patching the same thing? 
better should be doing
let data = [ ...this.state.data, item]; // just have reference to state, do not directly patch/update state. 
this.setState({data: data}) // then update the state once you're done locally.

File: src/InputTodo.js (hunk 3 above)
Comment (taleymirza) [reply src/InputTodo.js:4]: because mashhood bhai said so..

Comment (JavaEria) [reply src/InputTodo.js:4]: We were using a custom random generator which wasn't efficient as the lengths were not fixed, uuid provides timestamped version of random id of fixed length and is better for the id. @QaseemLodhi

File: src/InputTodo.js (hunk 4 above)
Comment (JavaEria) [reply src/InputTodo.js:13]: Can you explain why we need to store it in a local variable? @QaseemLodhi

Comment (JavaEria) [reply src/InputTodo.js]: @QaseemLodhi

File: src/EditTodoItem.js (hunk 1 above)
Comment (JavaEria) [reply src/EditTodoItem.js:6]: Can you give a reference of which linting should we use for React? @mashhoodr

File: src/InputTodo.js (hunk 4 above)
Comment (QaseemLodhi) [reply src/InputTodo.js:13]: easy for debugging, 
and its very minor but performant comparatively
easy to read and understand what the extracted value is.

File: src/InputTodo.js (hunk 3 above)
Comment (QaseemLodhi) [reply src/InputTodo.js:4]: @taleymirza, please avoid such kind of responses in the future and try to justify the answer.
@JavaEria that seems fine. but if we are not heavily concerned with generating alphanumeric id then for sake of simplicity and skipping dependencies we can just use Date.now().toString().

**Pull request merged**